src/main/java/org/dependencytrack/tasks/repositories/NugetMetaAnalyzer.java

```diff
@@ -111,7 +111,11 @@ private boolean performVersionCheck(final MetaModel meta, final Component compon
                     String responseString = EntityUtils.toString(response.getEntity());
                     var jsonObject = new JSONObject(responseString);
                     final JSONArray versions = jsonObject.getJSONArray("versions");
-                    final String latest = findLatestVersion(versions); // get the last version in the array
+
+                     // if the version is a pre-release version, we should not exclude pre-release versions
+                    final boolean excludePreRelease = component.getPurl().getVersion() != null && !component.getPurl().getVersion().contains("-"); 
```

Comment:
Latest version information is stored by repository type, namespace, and name of the component:

https://github.com/DependencyTrack/dependency-track/blob/af4461e86ae56eaee193017ec86ee2a690709932/src/main/java/org/dependencytrack/model/RepositoryMetaComponent.java#L53-L74

Thus, the check can't be conditional based on whether the analyzed component's current version is unstable. It would yield different results depending on whether a component with stable or unstable version was analyzed first.

The preference for including pre-release versions must be either global, or on a per-repository basis.

Reply:
Are you saying that the latest version is stored globally in dependency track and not against a project?


So in this scenario, the latest dependency is 1.1-release6 but the latest stable was 1.0

And we had 2 projects
Project A - used 1.0
Project B - used 1.1-release5

Then it would make a difference as to which project was scanned first? (Or last)